phnt/include/ntldr.h

```diff
@@ -883,163 +966,246 @@ NTSTATUS
 NTAPI
 LdrFindResource_U(
     _In_ PVOID DllHandle,
-    _In_ PLDR_RESOURCE_INFO ResourceInfo,
-    _In_ ULONG Level,
-    _Out_ PIMAGE_RESOURCE_DATA_ENTRY *ResourceDataEntry
+    _In_reads_(ResourceIdPathLength) LDR_EXTERNAL_DLL_PATH ResourceIdPath[],
```

Comment:
If we really want to eliminate `LDR_RESOURCE_INFO`. I'd change this to just be `PULONG_PTR` and leave out `LDR_EXTERNAL_DLL_PATH`.

Reply:
Indeed, each index has a separate meaning. I initially used `PULONG_PTR` but kept a helper type (with some adjustments to make it a union), but I got the impression that dmex didn't like the new type. I'll remove `LDR_EXTERNAL_DLL_PATH` in favor or `PULONG_PTR` and add back the helper type